js/script.js

```diff
@@ -99,13 +169,65 @@ document.querySelector('#book-form').addEventListener('submit', (e) => {
   const title = document.querySelector('#title').value;
   const author = document.querySelector('#author').value;
   const isbn = document.querySelector('#isbn').value;
+  const email = document.querySelector('#authorEmail').value;
+  const phone = document.querySelector('#author-phone').value;
 
   // Validate
-  if(title === '' || author === '' || isbn === '') {
-    UI.showAlert('Please fill in all fields', 'danger');
+  if (title === '') {
+    UI.show_error_text('.titleError');
+  }
+  if (author === '') {
+    UI.show_error_text('.authorError');
+  }
+  if (isbn === '') {
+    UI.show_error_text('.isbnError');
+  }
+  if (email === '') {
+    UI.show_error_text('.emailError');
+  }
+  if (phone === '') {
+    UI.show_error_text('.phoneError');
+  }
+
+  if(title === '' || author === '' || isbn === '' || email === '') {
+    return;
   } else {
+    // Check if the form is in update mode
+    if (document.querySelector('.btn-block').value === 'Update') {
+      const books = Store.getBooks();
+      let book = null;
+
+      books.forEach((b, index) => {
+        if(b.isbn === isbn) {
+          book = b;
+        }
+      });
+
+      book.title = title;
+      book.author = author;
+      book.isbn = isbn;
+      book.email = email;
+      book.phone = phone;
+
+      // Update book to store
+      Store.UpdateBook(book);
+
+      // Update book to UI
+      UI.clearBooks();
+      UI.displayBooks();
+
+      // Show success message
+      UI.showAlert('Book Updated', 'success');
+
+      // Clear fields
+      UI.clearFields();
+
+      document.querySelector('.btn-block').value = 'Add Book';
+      return;  
+    }
+    
     // Instatiate book
-    const book = new Book(title, author, isbn);
+    const book = new Book(title, author, email, phone, isbn);
 
     // Add Book to UI
     UI.addBookToList(book);
```

Comment:
consider removing this in favor of more generic 
```
UI.clearBooks();
UI.displayBooks();
```
after `Store.addBook(book)`

Advantages: less code to maintain, less potential for bugs, cleaner, UI is derived from state.
Catch: performance. Clearing and adding all rows again. (premature optimization?)